### PR TITLE
Migrate CI to GitHub Actions

### PR DESCRIPTION
**Pull request by VKTB**
Migrates the CI from Travis to GitHub Actions and updates the README status badge to use that of GitHub Actions. The status badge will start working once this PR gets merged in master, but this is the status badge for this branch to prove that it works:

[Build Status](https://github.com/icatproject/authn.db/actions?query=workflow%3A%22CI+Build%22) 

The CI on Travis was not running the integration tests so I configured the GitHub Actions CI to run these from now.

- [ ] Java 8 CI should pass
- [ ] Java 11 CI should pass
- [ ] Should run maven build
- [ ] Should run unit tests
- [ ] Should run integration tests

### CODE REVIEW

**Comment by VKTB** [comment]
This PR should not be merged until https://github.com/icatproject-contrib/icat-ansible/pull/72 is merged and the TODO below is addressed
https://github.com/icatproject/authn.db/blob/d1fb4a8a70b45241964df0ee6516c2f6a989fd7c/.github/workflows/ci-build.yml#L44-L45

**Comment by VKTB** [comment]
TODO comment was addressed in https://github.com/icatproject/authn.db/pull/9/commits/557871322cef9394c7db07da5c7e75654c7d98ef after https://github.com/icatproject-contrib/icat-ansible/pull/72 was merged so this PR can be merged as soon as it is approved.

**Review by patrick-austin — APPROVED** [review]
Looks good to me, old Travis tests didn't seem to be running the integration test(s) so good that this catches them.

File: .github/workflows/ci-build.yml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        run: |
+          grep payara ~/.bash_profile > payara_path_command
+          source payara_path_command
+          mvn install -DskipTests
+
+      - name: Run Unit Tests
+        run: mvn test -B
+
+      - name: Run Integration Tests
+        run: mvn failsafe:integration-test -B
```
Comment (patrick-austin) [inline .github/workflows/ci-build.yml:93]:
```suggestion
        run: |
          mvn failsafe:integration-test -B
          mvn failsafe:verify
```

**Review by patrick-austin — CHANGES_REQUESTED** [review]
See comments on https://github.com/icatproject/authn.ldap/pull/12

**Review by patrick-austin — APPROVED** [review]

**Pull request merged**